lib/scanner/walk_test.go

```diff
@@ -366,6 +366,43 @@ func TestWalkSymlinkWindows(t *testing.T) {
 	}
 }
 
+func TestWalkRootSymlink(t *testing.T) {
+
+	// Create a folder with a symlink to it
+
+	link := "link"
+
+	os.Remove(link)
+	defer os.Remove(link)
+
+	if err := osutil.DebugSymlinkForTestsOnly("testdata/dir1", link); err != nil && runtime.GOOS == "windows" {
+		// Probably we require permissions we don't have.
+		t.Skip("Need admin permissions or developer mode to run symlink test on Windows: " + err.Error())
+	}
```

Comment:
What if error non nil, but we not on Windows?

Reply:
I was following suite with the above referenced https://github.com/syncthing/syncthing/blob/445c4edeca8f70bec8ae02a17db234ed06e88d62/lib/config/config_test.go#L464. The only reason to check for the error is because of Window's unique behavior of requiring admin privileges or developer mode to create symlinks and this test is more a way to notify that the testing environment hasn't been set up correctly. Checking for errors on other platforms is testing an external dependency.

Reply:
Still, if this fails on non-windows, the test probably will fail anyway, so we should abort early.

Reply:
Actually, it looks like Go 1.4 added support for Windows symlinks - https://github.com/golang/go/commit/cf521ce64f50c4f300294d6b649f34eca87bb8a3. Scrolling through the commit it looks like they are also testing for long path names. I'll redactor that add a test.